### PR TITLE
HydroDyn user documentation update for large platform rotation

### PR DESCRIPTION
**Pull request by luwang00**
**Feature or improvement description**
* This PR updates the HydroDyn user documentation to reflect the recent input file changes for the modeling of large platform yaw motion. 
* It also updates error messages, variable descriptions, and comments in HydroDyn to match the finalized implementation of large platform rotation.

**Related issue, if one exists**
This is the documentation update for PR #2203

**Impacted areas of the software**
Docs, HydroDyn

**Test results, if applicable**
The changes have no impact on the results.

### CODE REVIEW

**Comment by luwang00** [comment]
@andrew-platt Looks like we have the wrong tag here.

**Comment by andrew-platt** [comment]
Whoops.  I was looking at something in _AeroDyn_ when I set the tag...

File: docs/source/user/hydrodyn/appendix.rst [hunk 1]
```diff
@@ -12,9 +12,13 @@ structure::
...
+                   1   ExctnMod       - Wave-excitation model {0: no wave-excitation calculation, 1: DFT, 2: state-space} (switch) [only used when PotMod=1; STATE-SPACE REQUIRES *.ssexctn INPUT FILE; if PtfmYMod=1, need ExctnMod=0 or 1]
                    0   ExctnDisp      - Method of computing Wave Excitation {0: use undisplaced position, 1: use displaced position, 2: use low-pass filtered displaced position) [only used when PotMod=1 and ExctnMod>0 and SeaState's WaveMod>0]} (switch)
                   10   ExctnCutOff    - Cutoff (corner) frequency of the low-pass time-filtered displaced position (Hz) [>0.0] [used only when PotMod=1, ExctnMod>0, and ExctnDisp=2]) [only used when PotMod=1 and ExctnMod>0 and SeaState's WaveMod>0]} (switch)
+                   0   PtfmYMod       - Model for large platform yaw offset {0: Static reference yaw offset based on PtfmRefY, 1: dynamic reference yaw offset based on low-pass filtering the PRP yaw motion with cutoff frequency PtfmYCutOff} (switch)
+                   0   PtfmRefY       - Constant (if PtfmYMod=0) or initial (if PtfmYMod=1) platform reference yaw offset (deg)
```
Comment (jjonkman) [inline docs/source/user/hydrodyn/appendix.rst:19]: Presumably this angle should be in the [-180, 180) deg range?

File: docs/source/user/hydrodyn/input_files.rst [hunk 2]
```diff
@@ -200,6 +213,82 @@ motion to prevent double counting the contributions from first-order
...
+in platform heading to filter out as much wave-frequency platform 
+motion as possible while minimizing the phase shift in the low-frequency 
+heading change. Throughout the simulation, the instantaneous 
+platform yaw rotation should stay within +/-15 deg of the now 
+time-dependent **PtfmRefY**. A severe warning will be displayed if 
+this requirement is not met at any point during the simulation, while 
+still allowing the simulation to continue if possible. 
+
+With **PtfmYMod** = 1, HydroDyn requires the first- and second-order 
+potential-flow wave excitation input file(s) to cover the full range
```
Comment (jjonkman) [inline docs/source/user/hydrodyn/input_files.rst:257]: I thought second-order must be disabled with PtfmYMod =1?

**Review by jjonkman — COMMENTED** [review]
Overall this looks great!  Please find a couple minor comments below.

File: docs/source/user/hydrodyn/input_files.rst (hunk 2 above)
Comment (luwang00) [reply docs/source/user/hydrodyn/input_files.rst:258]: We should still support mean- and slow-drift loads through `MnDrift` or `NewmanApp`. This is explained below, but I added a clarification here as well.

File: docs/source/user/hydrodyn/appendix.rst (hunk 1 above)
Comment (luwang00) [reply docs/source/user/hydrodyn/appendix.rst:19]: `PtfmRefY` does not need to be strictly between [-180,180), not intentionally at least. Integer multiples of 360 deg are added as appropriate at various places in the code to handle this.

**Review by andrew-platt — APPROVED** [review]

**Pull request merged**